core/document-types/page.php

```diff
@@ -89,46 +85,51 @@ public static function get_site_settings_url_config( $should_return_string_forma
 		];
 	}
 
-	public static function get_elementor_create_new_page_url( $active_tab = null ): string {
+	public static function get_create_new_editor_page_url( $active_tab = null ): string {
 		$active_kit_id = Plugin::$instance->kits_manager->get_active_id();
+		$args = [];
 
-		if ( empty( $active_kit_id ) ) {
-			return Plugin::$instance->documents->get_create_new_post_url( 'page' );
+		if ( ! empty( $active_kit_id ) ) {
+			$args['active-document'] = $active_kit_id;
 		}
 
-		$args = [
-			'active-document' => $active_kit_id,
-			'active-tab' => $active_tab,
-		];
+		if ( $active_tab ) {
+			$args['active-tab'] = $active_tab;
+		}
 
 		return add_query_arg( $args, Plugin::$instance->documents->get_create_new_post_url( 'page' ) );
 	}
 
-	private static function get_elementor_edit_url( int $post_id ): string {
+	private static function get_elementor_edit_url( int $post_id, $args = [] ): string {
 		$active_kit_id = Plugin::$instance->kits_manager->get_active_id();
-		$args = [
-			'active-document' => $active_kit_id,
-			'action' => 'elementor',
-			'post' => $post_id,
-		];
 
-		return add_query_arg( $args, get_admin_url() . 'post.php' );
+		if ( ! empty( $active_kit_id ) ) {
+			$args['active-document'] = $active_kit_id;
+		}
+
+		$page = new self( [ 'post_id' => $post_id ] );
+
+		return add_query_arg( $args, $page->get_edit_url() );
 	}
 
-	private static function get_elementor_page() {
-		if ( 'page' === get_option( 'show_on_front' ) ) {
-			$home_page_id = get_option( 'page_on_front' );
+	public static function get_elementor_page() {
+		return 'page' === get_option( 'show_on_front' )
+			? self::get_home_page()
+			: self::get_any_page();
+	}
 
-			return get_post( $home_page_id ) ?? null;
-		}
+	private static function get_home_page() {
+		$home_page_id = get_option( 'page_on_front' );
 
-		$pages = get_pages( [
+		return get_post( $home_page_id ) ?? null;
```

Comment:
Who says the home page was edited with Elementor? 🤔

Reply:
This is just the way it was in the home screen even before Mugur's change

Either way I removed it :)

Reply:
I see. Thanks 👊